src/index.js

```diff
@@ -79,9 +83,42 @@ export default class MarkersPlugin extends UICorePlugin {
       internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
     }
     internalMarker.onDestroy()
+    this._markers.splice(index, 1)
```

Comment:
I think calculating the index here is better as it reduces the chance of error. Or change this to `removeInternalMarker(index)` and the first line would be `internalMarker = this._markers[index]` to keep the performance benefit.

Currently there are 2 arguments to this that both depend on each other, which could both be supplied wrong and would cause strange results., so I think one argument that would be better.